skactiveml/pool/_typi_clust.py

```diff
@@ -0,0 +1,228 @@
+            indicated by self.missing_label)
+        candidates: None or array-like of shape (n_candidates), dtype = int or
+            array-like of shape (n_candidates, n_features), optional (default=None)
+            If candidates is None, the unlabeled samples from (X,y) are considered
+            as candidates
+            If candidates is of shape (n_candidates) and of type int,
+            candidates is considered as a list of the indices of the samples in (X,y).
+            If candidates is of shape (n_candidates, n_features), the candidates are
+            directly given in the input candidates (not necessarily contained in X)
+        batch_size: int, optional(default=1)
+            The number of samples to be selects in one AL cycle.
+        return_utilities: bool, optional(default=False)
+            If True, also return the utilities based on the query strategy
+
+        Returns
+        ----------
+        query_indices: numpy.ndarry of shape (batch_size)
+            The query_indices indicate for which candidate sample a label is
+            to queried, e.g., `query_indices[0]` indicates the first selected sample.
+            If candidates in None or of shape (n_candidates), the indexing
+            refers to samples in X.
+            If candidates is of shape (n_candidates, n_features), the indexing
+            refers to samples in candidates.
+        utilities: numpy.ndarray of shape (batch_size, n_samples) or
+            numpy.ndarray of shape (batch_size, n_candidates)
+            The utilities of samples for selecting each sample of the batch.
+            Here, utilities means the typicality in the considering cluster.
+            If candidates is None or of shape (n_candidates), the indexing
+            refers to samples in X.
+            If candidates is of shape (n_candidates, n_features), the indexing
+            refers to samples in candidates.
+        """
+        X, y, candidates, batch_size, return_utilities = self._validate_data(
+            X, y, candidates, batch_size, return_utilities, reset=True
+        )
+
+        X_cand, mapping = self._transform_candidates(candidates, X, y)
+
+        # Validate init parameter
+        if not issubclass(self.cluster_algo, ClusterMixin):
+            raise TypeError(
+                "Only clustering algorithm from super class sklearn.ClusterMixin is supported."
+            )
+
+        if not isinstance(self.k, int):
+            raise TypeError("Only k as integer is supported.")
+
+        if not isinstance(self.cluster_algo_param, dict):
+            raise TypeError(
+                "Please pass a dictionary with corresponding parameter name and value in the init function."
+            )
+
+        if not isinstance(self.n_cluster_param_name, str):
+            raise TypeError("n_cluster_param_name supports only string.")
+
+        selected_samples = labeled_indices(y, missing_label=self.missing_label)
```

Comment:
```suggestion
        labeled_sample_indices = labeled_indices(y, missing_label=self.missing_label)
```
Apply resulting changes in the following accordingly.